File: Java/097.java
import java.math.BigInteger;

public class Problem097 {

    public static void main(String[] args) {
        BigInteger base = BigInteger.valueOf(2);
        BigInteger exponent = BigInteger.valueOf(7830457);
        BigInteger multiplier = BigInteger.valueOf(28433);
        BigInteger modulus = BigInteger.valueOf(10_000_000_000L);

        // Calculate 2^7830457 % 10^10
        BigInteger powerOfTwoMod = base.modPow(exponent, modulus);

        // Calculate (28433 * (2^7830457 % 10^10)) % 10^10
        BigInteger term1 = multiplier.multiply(powerOfTwoMod).mod(modulus);

        // Add 1 and take modulus again
        BigInteger result = term1.add(BigInteger.ONE).mod(modulus);

        System.out.println(result);
    }
}

File: Rust/097.rs
use num_bigint::BigUint;

fn main() {
    let base = BigUint::from(2u32);
    let exponent = BigUint::from(7830457u32);
    let multiplier = BigUint::from(28433u32);
    let modulus = BigUint::from(10_000_000_000u64);

    // Calculate 2^7830457 % 10^10
    let power_of_two_mod = base.modpow(&exponent, &modulus);

    // Calculate (28433 * (2^7830457 % 10^10)) % 10^10
    let term1 = (&multiplier * &power_of_two_mod) % &modulus;

    // Add 1 and take modulus again
    let result = (&term1 + BigUint::from(1u32)) % &modulus;

    println!("{}", result);
}
